addons/binding/org.openhab.binding.oceanic/ESH-INF/thing/softener.xml

```diff
@@ -0,0 +1,154 @@
+			<channel id="pressure" typeId="pressure" />
+			<channel id="maxpressure" typeId="pressure" />
+			<channel id="minpressure" typeId="pressure" />
+			<channel id="maxflow" typeId="flow" />
+			<channel id="lastgeneration" typeId="time" />
+			<channel id="normalregenerations" typeId="number" />
+			<channel id="serviceregenerations" typeId="number" />
+			<channel id="incompleteregenerations" typeId="number" />
+			<channel id="allregenerations" typeId="number" />
+		</channels>
+
+		<config-description>
+			<parameter name="port" type="text" required="true">
+				<label>Serial Port</label>
+				<description>Serial Port of the openHAB host the Oceanic Water Softener is connected to</description>
+			</parameter>
+			<parameter name="interval" type="decimal" required="true">
+				<label>Polling Interval</label>
+				<description>Interval in seconds to poll the Oceanic Water Softener for status updates</description>
+				<default>60</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="alarm">
+		<item-type>String</item-type>
+		<label>Alarm</label>
+		<description>Current alarm description, if any</description>
+	</channel-type>
+	<channel-type id="alert">
+		<item-type>String</item-type>
+		<label>Alert</label>
+		<description>Current alert description, if any, to notify a shortage of salt</description>
+	</channel-type>
+	<channel-type id="flow" advanced="true">
+		<item-type>Decimal</item-type>
+		<label>Flow</label>
+		<description>Flow in l/min</description>
+	</channel-type>
+	<channel-type id="reserve" advanced="true">
+		<item-type>Decimal</item-type>
+		<label>Water Reserve</label>
+		<description>Water reserve in l before regeneration has to start</description>
+	</channel-type>
+	<channel-type id="cycle">
+		<item-type>String</item-type>
+		<label>Cycle</label>
+		<description>Indicates the stage of the regeneration cycle</description>
+	</channel-type>
+	<channel-type id="time" advanced="true">
+		<item-type>String</item-type>
+		<label>Date/Time</label>
+		<description>Date/Time stamp</description>
+	</channel-type>
+	<channel-type id="unit" advanced="true">
+		<item-type>String</item-type>
+		<label>Unit</label>
+		<description>Hardness unit used to express hardness</description>
+	</channel-type>
+	<channel-type id="hardness" advanced="true">
+		<item-type>Decimal</item-type>
+		<label>Water hardness</label>
+		<description>Water hardness expressed using the chosen hardness unit</description>
+	</channel-type>
+	<channel-type id="cylinderstate">
+		<item-type>String</item-type>
+		<label>Cylinder State</label>
+		<description>Indicates the state of the regeneration cylinder(s)</description>
+	</channel-type>
+	<channel-type id="salt">
+		<item-type>Decimal</item-type>
+		<label>Salt</label>
+		<description>Volume of salt remaining, in kg</description>
+	</channel-type>
+	<channel-type id="regeneratenow">
+		<item-type>Boolean</item-type>
```

Comment:
Nope. That does not exist. You probably wanted to use Switch. Same for the channels below.